lib/Agrammon/OutputFormatter/GUI.pm6

```diff
@@ -67,6 +56,48 @@ sub get-data($model, $outputs) {
     return @records;
 }
 
+sub make-record($fq-name, $output, $model, $raw-value, $var, $filter-id?) {
+    my $format = $model.output-format($fq-name, $output);
+    my $full-value = flat-value($raw-value);
+    my $value = ($format  && $full-value.defined) ?? sprintf($format, $full-value)
+                                                  !! $full-value;
+    my $filter;
+    if $filter-id {
+        $filter-id ~~ / '=' (.+) /;
+        $filter = ~$0;
+    }
+    return %(
+        :$format,
+        :print($model.output-print($fq-name, $output)),
+        :order($model.output-order($fq-name, $output)),
+        :labels($model.output-labels($fq-name, $output)),
+        :units($model.output-units($fq-name, $output)),
+        :fullValue($full-value),
+        :$value,
+        :$var,
+        :$filter,
+    );
+}
+
+sub push-filters(@records, $fq-name, $output, $model,
+                 Agrammon::Outputs::FilterGroupCollection $collection,
+                 $var) {
+    my @results = $collection.results-by-filter-group;
+    for @results {
```

Comment:
A straight `for $collection.results-by-filter-group {` would avoid a copy of the data into an array and out again (though it's probably not all that performance sensitive here).

Reply:
Done, not sure why I did that ...